Gameplay/entities/enemies/EnemyController.cpp

```diff
@@ -785,7 +791,7 @@ void Hachiko::Scripting::EnemyController::RegisterHit(int damage, float3 directi
 	}
 
 	_combat_stats->ReceiveDamage(damage);
-	game_object->ChangeEmissiveColor(float4(255, 255, 255, 255), 0.3f, true);
+	damage_effect_progress = damage_effect_duration;
```

Comment:
If damage_effect_duration is higher than 1.0 this won't work, we should be setting the effect process to 1.0 if we are using the decrement method (damage_effect_progress -= Time::DeltaTime() / damage_effect_duration) (i would make it reverse so it goes from 0 to 1 though).